### ISSUE TITLE
Not receiving notifications

### ISSUE THREAD

**Issue by Procyon-b** [JasonBarnabe/greasyfork#792]
A user has posted a question on one of my userstyle, but I didn't receive any notification.
My email address is on gmx.net, and I know by experience that their spam filtering with blacklisted IPs is brutal (no trace of rejected emails). Could this be the issue here? Or is it a bug?
`"Get an email when discussions are posted on your scripts or when those discussions get new comments"` is checked.

**Edit**
I've received a reply to a conversation, and notice that, while I know I hit "subscribe" a couple hours ago, it is now again displaying the conversation with "subscribe" instead of "unsubscribe".

**Edit** (5min later)
The user is online now and he replied immediately to my comment. The discussion is again unsubscribed (not listed in the `https://greasyfork.org/en/discussions?me=subscribed`).

**Comment by JasonBarnabe** [comment]
Can you let me know the URL of the style/discussion?

**Comment by Procyon-b** [comment]
401760-google-images-show-size/discussions/63018

**Comment by JasonBarnabe** [comment]
Thanks. So what exactly are you reporting? That you're not receiving emails, or some bug in the subscribed/not subscribed status?

(Note that choosing the "Get an email..." option in your user settings does _not_ affect the subscribed status you see when viewing the discussion.)

**Comment by Procyon-b** [comment]
First, I noticed that while I have the option checked, I don't receive notifications. Never received one.
It could be my email provider. My original question was: is it a bug or a problem with my email, or something else

While writing this issue, I subscribed to the new conversation that prompted me to come here. The word in the link on the page was then clearly "unsubscribe", and the conversation appeared in `/discussions?me=subscribed`.
When the user replied to that conversation :
- no notication received
- the conversation's page show "subscribe"
- doesn't appear in `/discussions?me=subscribed` anymore

I hit subscribe again; verified several times if it was still in the list; and 5min later when a reply came in, no subscribed anymore.

So probably 2 issues here:
- I don't receive notifications
- the "subscribe" option doesn't stick

**Comment by JasonBarnabe** [comment]
I can see that three emails were sent to you recently, corresponding to the times of the 1st, 3rd, and 5th comments in the discussion (all the ones not by you).

I don't of any mechanism by which you would be unsubscribed due to someone else replying. There is a checkbox when you reply that will change your subscribed status. Did you change that?

**Comment by Procyon-b** [comment]
**email**: this fits with gmx filtering the messages.
I don't remember how I created my account. Does it involve email verification at all? Or maybe I used my github account.

My current settings for notifications are:
- Get an email when discussions are posted on your scripts or when those discussions get new comments
- By default, subscribe to any discussions you start
- By default, subscribe to any discussions you comment in
- By default, subscribe to any conversations you start
- By default, subscribe to any conversations involving you that you do not start

**Comment by JasonBarnabe** [comment]
"Conversations" are private messages so are unrelated to this.

These settings:
- By default, subscribe to any discussions you start
- By default, subscribe to any discussions you comment in

Simply control the default value of the "Notify me of any replies" checkbox you see when posting. If you uncheck this box and submit a reply, you are unsubscribing.

**Comment by Procyon-b** [comment]
The thing is that I'm unsubscribed when the user replies, not when I post one.
I've subscribed once again. Post a reply and see if this change the setting.
I'm about to go offline (sleep time long overdue), but I'm still here for a few minutes...

**Comment by Procyon-b** [comment]
> * By default, subscribe to any discussions you start
> * By default, subscribe to any discussions you comment in
> 
> Simply control the default value of the "Notify me of any replies" checkbox you see when posting. If you uncheck this box and submit a reply, you are unsubscribing.

I thought I misunderstood your reply. But no.

I see no "Notify me of any replies" checkbox when replying to a discussion. Maybe because I'm supposed to receive notifications by default.
I see this line: `You will be notified of any replies due to your script notification settings.`

**Comment by Procyon-b** [comment]
And after reading my reply...

Maybe I'm unsubscribed because the setting is useless considering my global setting?

**Comment by JasonBarnabe** [comment]
Because you have "Get an email when discussions are posted on your scripts or when those discussions get new comments" enabled, the subscribe checkbox on reply didn't render. The code was considering it not being there (which should mean leave the subscription status alone) the same as it being unchecked. I've fixed that.

**Comment by Procyon-b** [comment]
A normal user having no problem receiving notification would not feel the need to subscribe to conversations on his own scripts.

By the way, I changed the email address and I'm now receiving emails (at least the "forgot your password" works).
I have checked your server's IP in several RBLs but all is clear with the lists tested. I don't know why gmx blocks these emails at that level (before filtering spam). Hotmail marks them as spam by default, but I can at least whitelist them.
